### DISCUSSION TITLE
How do I pass cmake parameter by vcpkg?

### DISCUSSION THREAD

**Discussion by tantan1** [microsoft/vcpkg#22096] [General]
How can I set some cmake parameter in vcpkg installation as we set in cmake-gui to add some options.

**Comment by Osyotr** [comment]
You can set some options in triplet file:
https://github.com/microsoft/vcpkg/blob/master/docs/users/triplets.md#variables

**Reply by tantan1** [reply]
For example, I want to install osgearth, it depends osg, osg has a optional package dependency Freetype and cmake option OSG_TEXT_USE_FONTCONFIG, FREETYPE_INCLUDE_DIR_freetype2 and FREETYPE_INCLUDE_DIR_ft2build, how can I config it? Can I config these just in one config file?

**Reply by Neumann-A** [reply]
Since those are not exposed via port features you have to use `--overlay-ports` for it and adjust the portfile yourself. (https://github.com/microsoft/vcpkg/blob/master/docs/specifications/ports-overlay.md)
